File: ios/Classes/ImaPlayerSettings.swift
//
//  ImaPlayerSettings.swift
//  ima_player
//
//  Created by gece on 26.12.2023.
//

import Foundation


struct ImaPlayerSettings {
    var isMuted: Bool = false
    var isMixed: Bool = false
    var autoPlay: Bool = true
    var showPlaybackControls: Bool = true
    
    var isAdsEnabled: Bool {
        get {
            return imaTag != nil && !imaTag!.isEmpty
        }
    }
    
    var imaTag: String?
    var videoUrl: String?
}

File: android/src/main/kotlin/dev/gece/imaplayer/ImaPlayerSettings.kt
package dev.gece.imaplayer

import android.net.Uri

class ImaPlayerSettings(mediaUrl: String, imaTag: String?) {
    var uri: Uri = Uri.parse(mediaUrl)
    var tag: Uri? = if (imaTag != null) Uri.parse(imaTag) else null
    var isMixed: Boolean = false;
    var autoPlay: Boolean = true
    var initialVolume: Double = 1.0
    var showPlaybackControls: Boolean = true;

    val isAdsEnabled get() = tag != null;

}